### DISCUSSION TITLE
Does supserset guest token support time grain?

### DISCUSSION THREAD

**Discussion by thanhpd56** [apache/superset#22388] [Q&A / Help]
Hi,
I trying to embed superset dashboard to web application through guest token, and it's working well. However I would like to set timegrain option while creating guest token instead of using native filter. Is it possible? 
Thanks.

**Comment by thanhpd56** [comment]
@rusackas please help. Thanks in advanced.

**Comment by rusackas** [comment]
Hi... I'm a little confused here on the intersection of the guest token (used for auth in embedded instances) and native filters (the dashboard feature, which affects charts). 

Native filters have a time _range_ filter type, but I'm not exactly sure what you're looking to do with time _grain_. If you can elaborate a bit more, hopefully I can either answer or bing some folks into the thread who can.

**Reply by thanhpd56** [reply]
Thanks for your help.
Actually I try to embed dashboard to my webapp, but don't want to use native filter, I want to use my own filter.
My filter include 2 options, the first one is time range and another is time grain.
I use time grain to compare difference between 2 period, for example between weeks, or months...
<img width="1003" alt="image" src="https://user-images.githubusercontent.com/3794898/207498604-8a27ccfb-e980-42e1-b765-20c71efe5685.png">

**Reply by rusackas** [reply]
Ahh, I think I understand your question now. You want to have a filter-ish  (since granularity is not a filter, exactly) control UI, in your own application, controlling Superset. 

The passing of filters to embedded instances of superset, from the host application, has been done by other orgs, but I'm not sure if the out-of-the-box embedded SDK supports this. Perhaps @lilykuang can help us here.

As for granularity as a filter... well... that's not a filter. I believe that's _not_ supported, just as changing a dimension (group by) or metric are also not filters and are not (currently) supported by the dashboard native filters implementation. Maybe @villebro or @zhaoyongjie can tell me if I'm mistaken and there's some way to accomplish this. Maybe there's a jinja trick or something ¯\\\_(ツ)_/¯

**Reply by rusackas** [reply]
Also, if I'm correct in understanding your question, the has nothing to do with the guest token, and the question should be re-titled something like "Does the Embedded SDK allow passing of granularity as a filter into a Superset dashboard?"
